### PR TITLE
1016 edit press form updates

### LINKED ISSUE #1016: Ability to modify press logo/cover image via manager
**Is your feature request related to a problem? Please describe.**
The default, material, and OLH templates display a `press_cover` (as an SVG?) but there doesn't appear to be an option to upload a new image for this in the press manager.

**Describe the solution you'd like**
An upload form in the press manager to set this image. Also, ideally, this would not be limited to SVG images but could accept other image formats.

### PR DESCRIPTION
**Pull request by ajrbyers**
Closes #1016

### CODE REVIEW

File: src/core/models.py [hunk 1]
```diff
@@ -651,6 +650,27 @@ def self_article_path(self):
...
             return os.path.join(settings.BASE_DIR, 'files', 'articles', str(self.article_id), str(self.uuid_filename))
 
+    def url(self):
+        from core.middleware import GlobalRequestMiddleware
+        request = GlobalRequestMiddleware.get_current_request()
+        url_kwargs = {'file_id': self.pk}
+
+        if request.journal and self.article_id:
+            return reverse(
+                'download_journal_file',
```
Comment (mauromsl) [inline src/core/models.py]: Does this work? The docs suggest the argument "viewname can be a URL pattern name or the callable view object"
So `journal_views.download_journal_file` or `"journal_file"`?

File: src/core/models.py [hunk 2]
```diff
@@ -651,6 +650,27 @@ def self_article_path(self):
...
+                kwargs=url_kwargs,
+            )
+        elif request.journal:
+            return reverse(
+                'download_journal_file',
```
Comment (mauromsl) [inline src/core/models.py]: As above. Also, I think this branch is a duplicate of the above

File: src/press/forms.py [hunk 3]
```diff
@@ -5,29 +5,46 @@
...
-    def __init__(self, *args, **kwargs):
-        super(PressForm, self).__init__(*args, **kwargs)
-        press = models.Press.objects.all()[0]
-        content_type = ContentType.objects.get_for_model(press)
-
-        self.fields['featured_journals'].queryset = journal_models.Journal.objects.filter(hide_from_press=False)
-        self.fields['carousel_news_items'].queryset = comms_models.NewsItem.objects.filter(
-            content_type=content_type,
-            object_id=press.pk
+        exclude = (
```
Comment (mauromsl) [inline src/press/forms.py]: Might be simpler and safer to write an include instead of an exclude

File: src/press/forms.py [hunk 4]
```diff
@@ -5,29 +5,46 @@
...
+        widgets = {
+            'featured_journals': forms.CheckboxSelectMultiple,
+        }
+
+    def save(self, commit=True):
+        press = super(PressForm, self).save(commit=False)
+        request = GlobalRequestMiddleware.get_current_request()
+
+        if press.thumbnail_image:
+            press.thumbnail_image.delete()
```
Comment (mauromsl) [inline src/press/forms.py]: Should defer the delete until the end, in case the below fails

File: src/press/forms.py [hunk 5]
```diff
@@ -5,29 +5,46 @@
...
+            'carousel_items', 'homepage_news_items', 'carousel',
+            'carousel_news_items', 'thumbnail_image',
         )
+        widgets = {
+            'featured_journals': forms.CheckboxSelectMultiple,
+        }
+
+    def save(self, commit=True):
+        press = super(PressForm, self).save(commit=False)
+        request = GlobalRequestMiddleware.get_current_request()
```
Comment (mauromsl) [inline src/press/forms.py]: This feels really clunky, we really need to refactor the files module.

File: src/core/homepage_elements/journals/templates/featured_journals.html [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    </div>
+    <div class="content">
+        <form method="POST">
+            {% csrf_token %}
+            {{ form|foundation }}
+            <button class="button success">Save</button>
+        </form>
+    </div>
+</div>
+{% endblock %}
```
Comment (mauromsl) [inline src/core/homepage_elements/journals/templates/featured_journals.html]: No new line at EOF

File: src/core/models.py [hunk 7]
```diff
@@ -651,6 +650,27 @@ def self_article_path(self):
         if self.article_id:
             return os.path.join(settings.BASE_DIR, 'files', 'articles', str(self.article_id), str(self.uuid_filename))
 
+    def url(self):
```
Comment (mauromsl) [inline src/core/models.py]: Maybe `download_url` would be more descriptive?

Comment (ajrbyers) [reply src/core/models.py]: It has to be URL for the Widget to work, but I do agree.

File: src/core/models.py (hunk 1 above)
Comment (ajrbyers) [reply src/core/models.py]: Raising NotImplementedError for these for the time being,

File: src/press/forms.py (hunk 5 above)
Comment (ajrbyers) [reply src/press/forms.py]: It does, and we should!

**Comment by ajrbyers** [comment]
I've addressed the comments as best I can :)

**Pull request merged**